### PR TITLE
Merge ign-gazebo6 ➡️  main (part 2)

### PR DESCRIPTION
**Pull request by chapulina**
# ➡️  Forward port

Port `ign-gazebo6 ` ➡️  `main`

Branch comparison: https://github.com/gazebosim/gz-sim/compare/main...ign-gazebo6

* Builds on top of https://github.com/gazebosim/gz-sim/pull/1630

@mjcarroll , I could use some help getting the changes from #1352  to play nicely with the reset stuff - #1390 is a nice reference.

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by chapulina** [comment]
CI result analysis:

* Actions Focal: one failed `INTEGRATION_odometry_publisher` and the other `INTEGRATION_scene_broadcaster_system` - they're flakes
* Actions Jammy: one failed `INTEGRATION_odometry_publisher` and the other `INTEGRATION_scene_broadcaster_system` - coincidentally the same flakes as in Focal
* Windows: No failures, just pre-existing warnings (109 is less than the [157 currently on `main`](https://build.osrfoundation.org/job/ign_gazebo-ci-win/220/))
* Homebrew: 
    * `UNIT_SimulationRunner_TEST` is a [know failure](https://build.osrfoundation.org/job/ignition_gazebo-ci-pr_any-homebrew-amd64/9028/testReport/(root)/UNIT_SimulationRunner_TEST/test_ran/history/)
    * `plugin_custom_sensor_system` - new failure caused by https://github.com/gazebosim/gz-sensors/pull/254 and being fixed in https://github.com/gazebosim/gz-sim/pull/1634

**Comment by mjcarroll** [comment]
I have on my list for this week:
* INTEGRATION_scene_broadcaster_system - I have most of a fix for this, just working out a last timing kink on all platforms.
* Windows: No failures, just pre-existing warnings - I have spun up a brand new Windows install to work through windows test failures and warnings throughout the stack.

**Comment by mjcarroll** [comment]
I have tested the reset examples locally and this looks like a clean merge, functionality is intact.  I don't see any weird interactions here.

File: src/SystemLoader.cc [hunk 1]
```diff
@@ -202,9 +194,29 @@ std::optional<SystemPluginPtr> SystemLoader::LoadPlugin(
...
+  return LoadPlugin(plugin);
+}
+
+//////////////////////////////////////////////////
+std::optional<SystemPluginPtr> SystemLoader::LoadPlugin(
+    const sdf::Plugin &_plugin)
+{
+  if (_plugin.Filename() == "" || _plugin.Name() == "")
+  {
+    ignerr << "Failed to instantiate system plugin: empty argument "
```
Comment (jennuine) [inline src/SystemLoader.cc:208]:
```suggestion
    gzerr << "Failed to instantiate system plugin: empty argument "
```

File: src/SystemManager.cc [hunk 2]
```diff
@@ -25,34 +25,37 @@ using namespace sim;
...
     entityCompMgr(_entityCompMgr),
     eventMgr(_eventMgr)
 {
+  transport::NodeOptions opts;
+  opts.SetNameSpace(_namespace);
+  this->node = std::make_unique<transport::Node>(opts);
+  std::string entitySystemService{"entity/system/add"};
+  this->node->Advertise(entitySystemService,
+      &SystemManager::EntitySystemAddService, this);
+  ignmsg << "Serving entity system service on ["
```
Comment (jennuine) [inline src/SystemManager.cc:40]:
```suggestion
  gzmsg << "Serving entity system service on ["
```

File: src/SystemManager.cc [hunk 3]
```diff
@@ -286,3 +292,40 @@ std::vector<SystemInternal> SystemManager::TotalByEntity(Entity _entity)
...
+void SystemManager::ProcessPendingEntitySystems()
+{
+  std::lock_guard<std::mutex> lock(this->systemsMsgMutex);
+  for (auto &req : this->systemsToAdd)
+  {
+    Entity entity = req.entity().id();
+
+    if (req.plugins().empty())
+    {
+      ignwarn << "Unable to add plugins to Entity: '" << entity
```
Comment (jennuine) [inline src/SystemManager.cc:316]:
```suggestion
      gzwarn << "Unable to add plugins to Entity: '" << entity
```

**Comment by chapulina** [comment]
> I have tested the reset examples locally and this looks like a clean merge, functionality is intact. I don't see any weird interactions here.

Thanks, @mjcarroll . I'll consider this an approval and merge this PR, we're way past feature freeze and need to branch off `gz-sim7` asap.

File: src/Util.cc [hunk 4]
```diff
@@ -427,15 +427,15 @@ void addResourcePaths(const std::vector<std::string> &_paths)
   char *ignPathCStr = std::getenv(systemPaths->FilePathEnv().c_str());
   if (ignPathCStr && *ignPathCStr != '\0')
   {
-    ignPaths = common::Split(ignPathCStr, ':');
+    ignPaths = common::Split(ignPathCStr, common::SystemPaths::Delimiter());
```
Comment (jennuine) [inline src/Util.cc:430]: Does `ign` need to be replaced with `gz` here? If not, we should update the above comment to say `Ignition file paths...` with a `Deprecated` note

File: src/rendering/SceneManager.cc [hunk 5]
```diff
@@ -2372,3 +2281,124 @@ AnimationUpdateData SceneManagerPrivate::ActorTrajectoryAt(
...
+    else if (extension == "dae")
+    {
+      // Load the mesh if it has not been loaded before
+      const common::Mesh *animMesh = nullptr;
+      if (!meshManager->HasMesh(animFilename))
+      {
+        animMesh = meshManager->Load(animFilename);
+        if (animMesh->MeshSkeleton()->AnimationCount() > 1)
+        {
+          ignwarn << "File [" << animFilename
```
Comment (jennuine) [inline src/rendering/SceneManager.cc:2340]:
```suggestion
          gzwarn << "File [" << animFilename
```

File: src/rendering/SceneManager.cc [hunk 6]
```diff
@@ -2372,3 +2281,124 @@ AnimationUpdateData SceneManagerPrivate::ActorTrajectoryAt(
...
+                  << std::endl;
+        }
+      }
+      animMesh = meshManager->MeshByName(animFilename);
+
+      // add the first animation
+      auto firstAnim = animMesh->MeshSkeleton()->Animation(0);
+      if (nullptr == firstAnim)
+      {
+        ignerr << "Failed to get animations from [" << animFilename
```
Comment (jennuine) [inline src/rendering/SceneManager.cc:2352]:
```suggestion
        gzerr << "Failed to get animations from [" << animFilename
```

File: src/rendering/SceneManager.cc [hunk 7]
```diff
@@ -2372,3 +2281,124 @@ AnimationUpdateData SceneManagerPrivate::ActorTrajectoryAt(
...
+        // collada loader loads animations with name that defaults to
+        // "animation0", "animation"1", etc causing conflicts in names
+        // when multiple animations are added to meshSkel.
+        // We have to clone the skeleton animation before giving it a unique
+        // name otherwise if mulitple instances of the same animation were added
+        // to meshSkel, changing the name that would also change the name of
+        // other instances of the animation
+        // todo(anyone) cloning is inefficient and error-prone. We should
+        // add a copy constructor to animation classes in ign-common.
+        // The proper fix is probably to update ign-rendering to allow it to
```
Comment (jennuine) [inline src/rendering/SceneManager.cc:2379]:
```suggestion
        // add a copy constructor to animation classes in gz-common.
        // The proper fix is probably to update gz-rendering to allow it to
```

**Review by jennuine — COMMENTED** [review]
I found some `ign` console loggers that should be updated. Should I open a separate PR for these?

**Comment by chapulina** [comment]
> I found some ign console loggers that should be updated. Should I open a separate PR for these?

Thanks, @jennuine ! Those are taken care of in #1633

**Pull request merged**